### PR TITLE
Use pre-computed distances in metrics

### LINKED ISSUE #7: Use waypoints and/or label centroids for pairwise distance metrics
**Is your feature request related to a problem? Please describe.**

Metrics that require calculating pairwise distances between cells (e.g. distance correlation, co-ranking) currently can't be calculated on even moderately large datasets due to time/memory limits. Alternative implementations may help somewhat but there are inherent limits to scalability of calculating pairwise distances.

**Describe the solution you'd like**

Instead of using all pairwise distances, these metrics could be calculated using selected waypoint cells and/or label centroids. The waypoint cells could be 5000-10000 cells randomly selected at the data processing stage and used for all methods/metrics. Calculating distances just to these cells should be significantly more scalable while still giving enough distances to calculate metric scores.

Using label centroids would be more efficient but gives information about a different level of structure which could be complementary to waypoint scores. 

**Describe alternatives you've considered**

Tried various approaches for calculating distances but none scale to our setting.

### PR DESCRIPTION
**Pull request by lazappi**
## Describe your changes

- Add calculation of distances to/between waypoints and label centroids to dataset pre-processing
- Add a post-processing component that calculates distances in the embedding space
- Modify co-ranking metrics to use pre-computed distances
- Modify distance correlation metrics to use pre-computed distances
- Add to centroid and between label distance correlation scores
- Move spectral distance correlation to a separate component
- Update documentation for distance correlation metrics
- Disable the trustworthiness metric as it is calculated as part of the co-ranking metrics

Fixes #7 

## Checklist before requesting a review

- [x] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [x] Major changes
  - [ ] Minor changes
  - [ ] Bug fixes

- [x] Proposed changes are described in the CHANGELOG.md

- [x] CI Tests succeed and look good!

### CODE REVIEW

**Comment by lazappi** [comment]
@rcannood The CI can't find the new test resources. I have synced them to S3, is there something else I need to do?

File: src/methods/simlr/script.R [hunk 1]
```diff
@@ -63,4 +63,5 @@ output <- anndata::AnnData(
   ),
   shape = input$shape
 )
+output$obs_names <- input$obs_names
```
Comment (rcannood) [inline src/methods/simlr/script.R:66]: FYI -- I think I sometimes solve this by:

```suggestion
  obs = input$obs[,integer(0)],
  var = input$var[,integer(0)]
)

**Review by rcannood — APPROVED** [review]
LGTM!

**Comment by rcannood** [comment]
> @rcannood The CI can't find the new test resources. I have synced them to S3, is there something else I need to do?

This is an issue with the caching -- Removing the cache resolved the issue.

**Pull request merged**